### PR TITLE
Ms af additem

### LINKED ISSUE #4: 4. As a user, I want to add a new item to my shopping list so I can start recording purchases.
A shopping list item consists of the following data points:

* Name of item
* How soon are you likely to buy it again?
  * Soon (in the next 7 days)
  * Kind of soon (in the next 14 days)
  * Not soon (in the next 30 days)
* Last purchased date

AC:

* User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
* When the user submits the form, the item is saved to the database, associated with the user’s token
* Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: `7` for “Soon”, `14` for “Kind of Soon”, and `30` for “Not Soon”
* The last purchased date should be set to `null` initially (i.e. you can create an item without purchasing it)
* Item names should be displayed as a simple, unordered list on the list view after they’re added

Notes:

* The add item form has the potential to be a lot of scope. The wireframe https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/add-item.png has several notes related to accessibility and UX.
* Eventually, the list will look like the following wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png
* It's good for your team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add all of those features.
* For this story, it's sufficient just to get a simple list of the items displaying.

### PR DESCRIPTION
**Pull request by Ginger-Mano**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

Users can access a form that allows them to enter the name of the item and select how soon they may need to buy it again (7, 14, or 30 days). When the user submits the form, they receive a digital token and the item they purchased will appear on the View List component. 

## Related Issue

Closes #4 

## Acceptance Criteria

User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)

When the user submits the form, the item is saved to the database, associated with the user’s token

Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)

Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓ | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓ | :link: Update dependencies |
| ✓ | :scroll: Docs              |

## Updates

**Front-end (Adding an item using form):**

<div align="center">
addItemrecord
</div>

<br>
<br>
**Back-end (Storing data to Cloud Firestore:**

<div align="center">
addItemFireStore
</div>

<br>
## Testing Steps / QA Criteria

**For UI changes:**
*From your terminal, pull down this branch with 'git pull origin ms-af-additem' 
*Check with 'git checkout ms-af-additem' 
*Type in 'bundle install' to see the newly added dependencies locally 
*Run 'npm start' to launch the app

**For Firestore changes:**
*To check if item is added to the "lists" collection in firestore, visit [here](https://console.firebase.google.com/u/0/project/tcl-3-smart-shopping-list/database/firestore/data~2Fitems~2FmXdjt73e7u9HTlvYLl3c)
*Click on "lists" tab under 'tcl-33-smart-shopping-list'
*Click through document list to find the item entered under 'add field' section

### CODE REVIEW

File: src/components/AddItem.js
Comment (laurenmbeatty) [inline src/components/AddItem.js:18]: Instead of adding to a collection called `lists`, add it to the collection that uses the users `token`. You'll receive that as a prop.

**Review by JenniferToops — APPROVED** [review]

File: src/Pages/ViewList.js
Comment (laurenmbeatty) [inline src/Pages/ViewList.js:6]: Eventually we'll pull in the `token` as a prop to `ViewList` and put it here:
```suggestion
  const [value, loading, error] = useCollection(db.collection(token));
```

File: src/components/AddItem.js
Comment (laurenmbeatty) [inline src/components/AddItem.js:44]: Excellent work here!

File: src/components/AddItem.js
Comment (laurenmbeatty) [inline src/components/AddItem.js:17]: Oh, what does this do?

**Review by laurenmbeatty — APPROVED** [review]
Great work you two! I'm defaulting to approval, but you may want to wait until the other team's PR is merged into main, then pull their work in. Otherwise there will be another follow-up PR.

File: src/Pages/ViewList.js
Comment (laurenmbeatty) [inline src/Pages/ViewList.js:5]:
```suggestion
const ViewList = ({ token }) => {
```

File: src/components/AddItem.js
Comment (laurenmbeatty) [inline src/components/AddItem.js:9]:
```suggestion
export default function AddItem({ token }) {
```

**Pull request merged**
